homework/src/index.js

```diff
@@ -31,17 +33,124 @@
+      showTableRepo(leftSection, data[theEventIndex]);
+    });
+  }
+
+  
+
+  // drow the table
+function showTableRepo(leftSection, repository) {
+  // create table of the contents
+  const lTable = createAndAppend('table', leftSection);
+  // line one
+  const tr1 = createAndAppend('tr', lTable);
+  createAndAppend('td', tr1, {html: 'Repository :', class: 'lable'});
+  const td1 = createAndAppend('td', tr1);
+  const nameLink = '<a href="' + repository.html_url + '" target="_blank">' + repository.name + '</a>';
+  createAndAppend('p', td1, {html: nameLink });
+  // line two
+  const tr2 = createAndAppend('tr', lTable);
+  createAndAppend('td', tr2, {html: 'Description :', class: 'lable'});
+  const td2 = createAndAppend('td', tr2);
+  createAndAppend('p', td2, {html: repository.description});
+  // line three
+  const tr3 = createAndAppend('tr', lTable);
+  createAndAppend('td', tr3, {html: 'Forks :', class: 'lable'});
+  const td3 = createAndAppend('td', tr3);
+  createAndAppend('p', td3, {html: repository.forks});
+  // line four
+  const tr4 = createAndAppend('tr', lTable);
+  createAndAppend('td', tr4, {html: 'Updated :', class: 'lable'});
+  const td4 = createAndAppend('td', tr4);
+  const defaultElDate = repository.updated_at;
+  const defaultDate = new Date(defaultElDate);
+  const endDate = defaultDate.toLocaleString();
+  createAndAppend('p', td4, {html: endDate});
+}
+
+
+
+  // function who creates a right content section
+  function contentRight(container, data, select) {
+    const rightSection = createAndAppend('div', container, { id: 'rightSection', class: 'whiteframe' });
+    createAndAppend('h5', rightSection, {html: "Contributions :", class: 'contributionsLable'});
+    const contributorsURL = data[0].contributors_url;
+    const ContributorsPromise = fetchJSON(contributorsURL);
+    ContributorsPromise
+      .then(contributors => {
+        showContributors(rightSection, contributors);
+      })
+      .catch(error => {
+        createAndAppend('div', rightSection, { html: error.message, class: 'alert-error' });
+      });
+
+    select.addEventListener('change', (event) => {
+      rightSection.innerHTML = "";
+      createAndAppend('h5', rightSection, {html: "Contributions :", class: 'contributionsLable'});
+      const theEventIndex = event.target.value;
+      const contributorsURL = data[theEventIndex].contributors_url;
+      const ContributorsPromise = fetchJSON(contributorsURL);
+      ContributorsPromise
+      .then(contributors => {
+        showContributors(rightSection, contributors);
+      })
+      .catch(error => {
+        createAndAppend('div', rightSection, { html: error.message, class: 'alert-error' });
+      });
     });
+
   }
 
+
+
+
+
+function showContributors(rightSection, contributors) {
+  contributors.forEach(contributor => {
+    const contrSection = createAndAppend('section', rightSection);
+    const contrMainLink = createAndAppend('a', contrSection, {href:  contributor.html_url});
```

Comment:
In terms of naming, all these `contr` prefixes are unnecessary, as in the context of this function everything evolves around contributors.